### ISSUE TITLE
JDownloader Homepage breaks with Basic Auth

### ISSUE THREAD

**Issue by rix1337** [causefx/Organizr#1830]
###### Operating System: Unraid / Docker
<hr>

##### Problem Description:

It is possible to enable Basic Authentification in the JDownloader API of FeedCrawler
Because there currently is no way to set username/password in the Homepage item's section I tried to add a URL like
`http://user:pass@192.168.0.1`

![grafik](https://user-images.githubusercontent.com/9930448/166113223-91fe9ba5-35eb-4e46-b757-dac701fcb0b2.png)
![grafik](https://user-images.githubusercontent.com/9930448/166113237-116759ec-242d-41be-ae47-20b12496477a.png)

Sadly this has not worked for me and yields an error.

The URL including user and pass works with curl and inside webbrowsers:
![grafik](https://user-images.githubusercontent.com/9930448/166113253-1582aeba-af19-4208-bb5a-0d1fc543e60b.png)
![grafik](https://user-images.githubusercontent.com/9930448/166113267-a22675f4-89f3-40d3-a7d7-481a0b7560f0.png)

<hr>

##### Reproduction Steps:

Set up Username and Password in FeedCrawler.
Set URL with user:pass@IP
Test URL
<hr>

#### Errors on screen?  If so paste here:

![grafik](https://user-images.githubusercontent.com/9930448/166113297-e8c46875-9ab6-4fbb-86ae-1a4ccd43ab7a.png)

**Comment by causefx** [comment]
lemme do some testing.

**Comment by causefx** [comment]
So this is using your feedcrawler package, and not the organizr standalone version - correct?

**Comment by rix1337** [comment]
Yeah, but I am planning to roll out basic auth to the standalone, if we can get this working

**Comment by causefx** [comment]
can you send me a config ini file so i can test on my end.

**Comment by rix1337** [comment]
@causefx to make things easier I have just implemented the basic auth feature in [the standalone version (MyJD-API)](https://github.com/rix1337/MyJD-API).

If you run 1.0.0 locally you can use the `--username` and `--password` parameters to set basic auth credentials.
The docker image also now runs 1.0.0 but will not handle these new parameters, yet. You can however edit the ini at /config to yield the same result.

If it is possible to integrate the basic auth feature in Organizr, I'll later update the docker image as well.

**Comment by causefx** [comment]
i'm still working on it

**Comment by rix1337** [comment]
Issue persists

**Comment by rix1337** [comment]
@causefx I just sent a Pull Request that should fix this.

**Comment by rix1337** [comment]
@causefx  Thanks for releasing the PR. I found some further issues that need adressing.

I have updated both MyJD-API as well as FeedCrawler to properly handle missing trailing slashes.

For some reason the PHP Requests library swallows trailing slashes, which my tools both expect, resulting in a 404.

That 404 has now been resolved on my end, leading to another error in the way my last PR handles authentification.

I have now locally debugged and fixed this as well. Will be sending over a PR with the fix soon.
